ssm_jax/linear_gaussian_ssm/models.py

```diff
@@ -455,25 +435,53 @@ def emap_step(_params):
         
         self.params = _params
         return jnp.array(log_probs)
+    
+    def fit_sgd(self, batch_emissions,
+                optimizer=optax.adam(1e-3),
+                batch_size=1,
+                num_epochs=50,
+                shuffle=False,
+                key=jr.PRNGKey(0),
+                **batch_inputs):
+        """Fit LinearGaussianSSM by running SGD on the marginal log likelihood
+        
+        The batch_emissions is initially of shape (N, T),
+        where N is the number of independent sequences and T is the length of a sequence.
+        Then, a random subset with shape (B, T) of entire sequence, not time steps,
+        is sampled at each step, where B is batch size.
 
-    @property
-    def hyperparams(self):
-        """Helper property to get a PyTree of model hyperparameters."""
-        return tuple()
-
-    # Use the to/from unconstrained properties to implement JAX tree_flatten/unflatten
-    def tree_flatten(self):
-        children = self.unconstrained_params
-        aux_data = self.hyperparams
-        return children, aux_data
-
-    @classmethod
-    def tree_unflatten(cls, aux_data, children):
-        return cls.from_unconstrained_params(children, aux_data)
+        Args:
+            batch_emissions (chex.Array): Independent sequences.
+            optimizer (optax.Optimizer): Optimizer.
+            batch_size (int): Number of sequences used at eash update step
+            num_eopchs (int): Iterations made through entire dataset.
+            shuffle (bool): Indicates whether to shuffle minibatches.
+            key (chex.PRNGKey): RNG key to shuffle minibatches.
+            
+        Returns:
+            losses: Output of loss_fn stored at each step.
+        """
+        def _loss_fn(params, minibatch_emissions, **minibatch_inputs):
+            self.unconstrained_params = params
+            scale = len(batch_emissions) / len(minibatch_emissions)
+            minibatch_lls = vmap(self.marginal_log_prob)(minibatch_emissions, **minibatch_inputs)
+            lp = self.log_prior() + minibatch_lls.sum() * scale
+            return - lp/batch_emissions.size
+        
+        params, losses = run_sgd(_loss_fn,
+                                 self.unconstrained_params,
+                                 batch_emissions,
+                                 optimizer=optimizer,
+                                 batch_size=batch_size,
+                                 num_epochs=num_epochs,
+                                 shuffle=shuffle,
+                                 key=key,
+                                 **batch_inputs)
+        self.unconstrained_params = params
+        return losses
     
     def _join_matrix(self, F, B, b, indicator):
```

Comment:
Add a comment explaining that if `bias_indicator=True` you add the bias vector `b` to the last column of the [F, B] matrix, so they can be jointly estimated.